docs/ref/contrib/postgres/fields.txt

```diff
@@ -916,3 +916,44 @@ types.
     .. attribute:: range_type
 
         The psycopg2 range type to use.
+
+Expression support for range boundaries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. class:: RangeBoundary(lower=LOWER_INC, upper=UPPER_EXC)
+
+For use together with
+:class:`django.contrib.postgres.constraints.ExclusionConstraint`, a simple
+utility class is provided that provides an easier way to write the range
```

Comment:
As implied above, I feel that this is too complex and only provides benefit for this one use case. I'm not sure it is easier than a two character string which is suitably validated...